### PR TITLE
Upgrade webpack dev server and add typescript dependencies

### PR DESCRIPTION
**Pull request by edleeks87**
# Description

Upgrades dev-server dependency and jest-cli dependency. Adds the typescript babel preset and configures the loader fr ts/x extensions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +35,7 @@
     "file-loader": "1.1.11",
     "identity-obj-proxy": "3.0.0",
     "jest": "23.4.1",
-    "jest-cli": "23.4.1",
+    "jest-cli": "^23.6.0",
```
Comment (andrewjtait) [inline package.json]: why'd this get bumped?

File: webpack.config.js [hunk 2]
```diff
@@ -132,6 +134,8 @@ module.exports = function(opts) {
   config.module = {
     rules: [
       babelLoader,
+      awesomeTypescriptLoader,
+      sourceMapLoader,
```
Comment (andrewjtait) [inline webpack.config.js]: where are these 2 loaders defined?

Comment (edleeks87) [reply webpack.config.js]: They should of been removed I'll push it up again

File: package.json (hunk 1 above)
Comment (edleeks87) [reply package.json]: There were warnings highlighted when I was installing and upgrading the other dependencies, I'll knock it back down if you don't want it bumping for now

File: jest.conf.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
   "snapshotSerializers": ["enzyme-to-json/serializer"],
   "testMatch": [
-    "**/__spec__.js",
-    "**/*.spec.js"
+    "**/__spec__.(ts|tsx|js)",
+    "**/*.spec.(ts|tsx|js)"
```
Comment (andrewjtait) [inline jest.conf.json]: we likely to write specs with typescript?

File: jest.conf.json [hunk 4]
```diff
@@ -19,5 +19,14 @@
...
     }
+  },
+  "moduleFileExtensions": [
+    "ts",
+    "tsx",
+    "js", 
+    "jsx"
+  ],
+  "transform": {
+    "^.+\\.(js|jsx|ts|tsx)$": "babel-jest"
```
Comment (andrewjtait) [inline jest.conf.json]: do you still need this transform config? thought you said you didn't any more

File: package.json [hunk 5]
```diff
@@ -18,6 +18,7 @@
     "@babel/polyfill": "7.2.5",
     "@babel/preset-env": "7.2.3",
     "@babel/preset-react": "7.0.0",
+    "@babel/preset-typescript": "^7.1.0",
```
Comment (andrewjtait) [inline package.json]: please use specific versions for all of these to avoid unexpected upgrades

File: package.json [hunk 6]
```diff
@@ -40,9 +41,14 @@
     "optimize-css-assets-webpack-plugin": "4.0.3",
     "sass-loader": "7.0.3",
     "style-loader": "0.21.0",
+    "tsconfig-paths-webpack-plugin": "^3.2.0",
+    "tslint": "^5.12.1",
+    "tslint-config-prettier": "^1.17.0",
```
Comment (andrewjtait) [inline package.json]: do we need this? we don't use prettier currently

File: package.json [hunk 7]
```diff
@@ -40,9 +41,14 @@
     "optimize-css-assets-webpack-plugin": "4.0.3",
     "sass-loader": "7.0.3",
     "style-loader": "0.21.0",
+    "tsconfig-paths-webpack-plugin": "^3.2.0",
```
Comment (andrewjtait) [inline package.json]: do we need this since you get set `paths` config?

File: tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            "log",
+            "time",
+            "timeEnd",
+            "trace"
+        ]
+    }
+  },
+  "jsRules": {
+    "max-line-length": {
+        "options": [120]
```
Comment (andrewjtait) [inline tsconfig.json]: whats the difference between this and the same rule defined above?

File: tslint.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+      "coverage/lcov-report/*.js"
+    ]
+  },
+  "rules": {
+    "interface-name" : [true, "never-prefix"],
+    "member-access": [true, "no-public"],
+    "ordered-imports": [false],
+    "object-literal-sort-keys": [false],
+    "class-name": false
+  }
```
Comment (andrewjtait) [inline tslint.json]: I wonder if we need some synchronisation between our eslint and tslint rules?

**Pull request merged**